src/main/java/com/redhat/exhort/providers/GradleProvider.java

```diff
@@ -0,0 +1,341 @@
+import org.tomlj.TomlParseResult;
+import org.tomlj.TomlTable;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static com.redhat.exhort.impl.ExhortApi.debugLoggingIsNeeded;
+
+/**
+ * Concrete implementation of the {@link Provider} used for converting dependency trees
+ * for Gradle projects (gradle.build) into a content Dot Graphs for Stack analysis or Json for
+ * Component analysis.
+ **/
+public final class GradleProvider extends BaseJavaProvider {
+
+  private Logger log = LoggersFactory.getLogger(this.getClass().getName());
+
+  public GradleProvider() {
+    super(Type.GRADLE);
+  }
+
+  @Override
+  public Content provideStack(final Path manifestPath) throws IOException {
+    Path tempFile = getDependencies(manifestPath);
+    if (debugLoggingIsNeeded()) {
+      String stackAnalysisDependencyTree = Files.readString(tempFile);
+      log.info(String.format("Package Manager Maven Stack Analysis Dependency Tree Output: %s %s", System.lineSeparator(), stackAnalysisDependencyTree));
+    }
+    Map<String, String> propertiesMap = extractProperties(manifestPath);
+
+    var sbom = buildSbomFromTextFormat(tempFile, propertiesMap, "runtimeClasspath");
+    var ignored = getIgnoredDeps(manifestPath);
+
+    return new Content(sbom.filterIgnoredDeps(ignored).getAsJsonString().getBytes(), Api.CYCLONEDX_MEDIA_TYPE);
+  }
+
+  private List<String> getIgnoredDeps(Path manifestPath) throws IOException {
+    List<String> buildGradleLines = Files.readAllLines(manifestPath);
+    List<String> ignored = new ArrayList<>();
+
+    var ignoredLines = buildGradleLines.stream()
+      .filter(this::isIgnoredLine)
+      .map(this::extractPackageName)
+      .collect(Collectors.toList());
+
+    // Process each ignored dependency
+    for (String dependency : ignoredLines) {
+      String ignoredDepInfo;
+      if (isNotation(dependency)) {
+        ignoredDepInfo = getDepFromNotation(dependency, manifestPath);
+      } else {
+        ignoredDepInfo = getDepInfo(dependency);
+      }
+
+      if (ignoredDepInfo != null) {
+        ignored.add(ignoredDepInfo);
+      }
+    }
+
+    return ignored;
+  }
+
+  private String getDepInfo(String dependencyLine) {
+    if (dependencyLine.contains("group: ")) {
+      // Process line with group: syntax
+      String[] parts = dependencyLine.split("'");
+      String groupId = parts[1].trim();
+      String artifactId = parts[3].trim();
+      String version = parts[5].trim();
+
+      PackageURL ignoredPackageUrl = toPurl(groupId, artifactId, version);
+      return ignoredPackageUrl.getCoordinates();
+    } else if (dependencyLine.contains("\"")) {
```

Comment:
basically it could be also single quote, not only double quotes 
for example:
```gradle
lugins {
    id 'java'
    id 'application'
}

mainClassName = 'App'

dependencies {
    compile 'com.google.guava:guava:23.0'

    testCompile 'junit:junit:4.12'
}

repositories {
    jcenter()
}
```

Reply:
done